service/controller.go

```diff
@@ -930,17 +930,24 @@ func validateVolSize(cr *csi.CapacityRange) (int64, error) {
 	}
 
 	var (
-		sizeGiB int64
-		sizeKiB int64
-		sizeB   int64
+		sizeGiBFloat float64
+		sizeGiB      int64
+		sizeKiB      int64
+		sizeB        int64
 	)
+
 	// VxFlexOS creates volumes in multiples of 8GiB, rounding up.
 	// Determine what actual size of volume will be, and check that
 	// we do not exceed maxSize
-	sizeGiB = minSize / kiBytesInGiB
+	// Calculate size in GiB using float for precision
+	sizeGiBFloat = float64(minSize) / float64(kiBytesInGiB)
+
+	// Use math.Ceil to round up to the nearest whole GiB
+	sizeGiB = int64(math.Ceil(sizeGiBFloat))
+
 	// if the requested size was less than 1GB, set the request to 1GB
 	// so it can be rounded to a 8GiB boundary correctly
-	if sizeGiB == 0 {
+	if sizeGiB < 1 {
```

Comment:
As these checks are focused on block storage, I am particularly interested in having similar validations for NFS as well, especially with the same input sizes. That how it behaves, I think should work but better to double check on this as well.

<img width="324" alt="image" src="https://github.com/user-attachments/assets/bd5bf427-be11-4174-86bd-b9766e2df653">

Reply:
I have tested NFS with filesystem sizes of 48.8 GiB and 3.001 GiB (3073 MiB). In both cases, the filesystems were successfully created. The only requirement for filesystem creation is that the size must not be less than 3 GiB, which is handled in the code.